pkg/k8s/commands/run.go

```diff
@@ -73,21 +76,61 @@ func run(ctx context.Context, opt cmd.Option, cluster string, artifacts []*artif
 	if err != nil {
 		return xerrors.Errorf("k8s scan error: %w", err)
 	}
+	workloadReport, rbacReport := separateMisConfigRoleAssessment(r, opt.ReportOption)
+
+	if err := report.Write(workloadReport, report.Option{
+		Format:        opt.Format,
+		Report:        opt.KubernetesOption.ReportFormat,
+		Output:        opt.Output,
+		Severities:    opt.Severities,
+		ColumnHeading: report.ColumnHeading(opt.ReportOption, report.WorkloadColumns()),
+	}); err != nil {
+		return xerrors.Errorf("unable to write workload results: %w", err)
+	}
 
-	if err := report.Write(r, report.Option{
-		Format:     opt.Format,
-		Report:     opt.KubernetesOption.ReportFormat,
-		Output:     opt.Output,
-		Severities: opt.Severities,
+	if err := report.Write(rbacReport, report.Option{
+		Format:        opt.Format,
+		Report:        opt.KubernetesOption.ReportFormat,
+		Output:        opt.Output,
+		Severities:    opt.Severities,
+		ColumnHeading: report.ColumnHeading(opt.ReportOption, report.RoleColumns()),
 	}); err != nil {
-		return xerrors.Errorf("unable to write results: %w", err)
+		return xerrors.Errorf("unable to write rbac results: %w", err)
 	}
 
 	cmd.Exit(opt, r.Failed())
 
 	return nil
 }
 
+func separateMisConfigRoleAssessment(k8sReport report.Report, rp option.ReportOption) (report.Report, report.Report) {
+	workloadMisconfig := make([]report.Resource, 0)
+	rbacAssessment := make([]report.Resource, 0)
+	for _, misConfig := range k8sReport.Misconfigurations {
+		if slices.Contains(rp.SecurityChecks, types.SecurityCheckRbac) && rbacResource(misConfig) {
+			rbacAssessment = append(rbacAssessment, misConfig)
+		} else {
+			if slices.Contains(rp.SecurityChecks, types.SecurityCheckConfig) && !rbacResource(misConfig) {
+				workloadMisconfig = append(workloadMisconfig, misConfig)
+			}
+		}
```

Comment:
this will not work in this case , as it will add rbac resources to misconfig columns :
```shell 
trivy  k8s --security-checks=config --report summary cluster
```